karapace/schema_record_producer.py

```diff
@@ -0,0 +1,161 @@
+from kafka import errors as kafka_errors, KafkaProducer
+from kafka.producer.future import FutureRecordMetadata
+from karapace.compatibility import CompatibilityModes
+from karapace.config import Config
+from karapace.errors import SchemaTooLargeException
+from karapace.key_format import KeyFormatter
+from karapace.offsets_watcher import OffsetsWatcher
+from karapace.schema_models import SchemaType, TypedSchema
+from karapace.typing import Subject, Version
+from karapace.utils import json_encode, KarapaceKafkaClient
+from typing import Optional, Union
+
+import logging
+import time
+
+LOG = logging.getLogger(__name__)
+
+
+class SchemaRecordProducer:
+    def __init__(self, config: Config, key_formatter: KeyFormatter, offsets_watcher: OffsetsWatcher) -> None:
+        self.kafka_timeout = 10
+
+        self.config = config
+        self.producer = self._create_producer()
+        self.key_formatter = key_formatter
+        self.offsets_watcher = offsets_watcher
+
+    def _create_producer(self) -> KafkaProducer:
```

Comment:
I know this is just refactoring of the code, but how much work it would be to switch using aiokafka here? Quick look seemed to be that all these messages are already produced from async methods as caller.

Reply:
I'd recommend not to refactor more in this PR.